src/compiler/program.ts

```diff
@@ -2730,6 +2735,9 @@ namespace ts {
                     createDiagnosticForOptionName(Diagnostics.Option_0_cannot_be_specified_without_specifying_option_1_or_option_2, "tsBuildInfoFile", "incremental", "composite");
                 }
             }
+            else if (options.incremental && !options.outFile && !options.out && !options.configFilePath) {

```

Comment:
shouldn't this also be checking `options.tsBuildInfoFile`?

Reply:
Thats absent since its in the else portion of `if (options.tsBuildInfoFile)`

Reply:
Ah, K. Would still be nice to see a compiler test with both flags.